politeiad/backendv2/tstorebe/plugins/pi/pi.go

```diff
@@ -23,6 +26,35 @@ var (
 	_ plugins.PluginClient = (*piPlugin)(nil)
 )
 
+// statusData is stored in an in-memory cache, it includes metadata which
+// won't change in the future to avoid expensive real-time evaluations.
+//
+// proposalStatus is optional and will be populated only if the proposal
+// status is not expected to change.
+//
+// voteStatus is optional and will be populated only for proposals with
+// vote status 'approved' to improve performance of retrieving proposal
+// statuses of approved proposals.
+type cacheData struct {
+	proposalStatus *pi.PropStatusT
+	voteStatus     *ticketvote.VoteStatusT
+}
+
+// piCacheLimit limits the number of entries in the piCache context.
+const piCacheLimit = 1000
+
+// piCache is used to cache proposals information that won't change in the
+// the future, and would require fetching the entire tlog tree in runtime.
+//
+// Number of entries stored in cache is limited. If the cache is full and a
+// new entry is being added, the oldest entry is removed from the `data`
+// map and the `entries` list.
+type piCache struct {
```

Comment:
Bad naming.  `piCache` and `cacheData` doesn't tell me anything about the data.  Am I to assume that there is never going to be any other data that is cached in the future?  That's what this naming implies.